torch/_inductor/lowering.py

```diff
@@ -2405,6 +2405,27 @@ def require_channels_last(_, *args, **kwargs):
     return args, kwargs
 
 
+def constrain_to_fake_tensors(args, kwargs, fake_args, fake_kwargs):
+    def apply_constraint(arg, fake_arg):
+        if isinstance(arg, ir.IRNode):
+            meta_stride_expr = [
+                s.node.expr if isinstance(s, torch.SymInt) else s
+                for s in fake_arg.stride()
+            ]
+            return ir.ExternKernel.require_exact_strides(arg, meta_stride_expr)
```

Comment:
Do we want to do storage_offset now or later?

Maybe better name for this api could be require_exact_layout so that we can add optional storage offset later

Reply:
Hmm, this doesn't handle storage_offset now. so i guess later